packages/contracts/contracts/router/modules/exchanges/NFTXModule.sol

```diff
@@ -0,0 +1,226 @@
+    // --- Fields ---
+
+    INFTXMarketplaceZap public constant NFTX_MARKETPLACE =
+        INFTXMarketplaceZap(0x0fc584529a2AEfA997697FAfAcbA5831faC0c22d);
+
+    bytes4 public constant ERC721_INTERFACE = 0x80ac58cd;
+    bytes4 public constant ERC1155_INTERFACE = 0xd9b67a26;
+
+    // --- Constructor ---
+
+    constructor(address owner, address router)
+        BaseModule(owner)
+        BaseExchangeModule(router)
+    {}
+
+    // event LogStr(string message);
+
+    // --- Fallback ---
+
+    receive() external payable {}
+
+   
+    // --- Multiple ETH listings ---
+
+    function buyWithETH(
+        INFTXMarketplaceZap.BuyOrder[] calldata orders,
+        ETHListingParams calldata params,
+        Fee[] calldata fees
+    )
+        external
+        payable
+        nonReentrant
+        refundETHLeftover(params.refundTo)
+        chargeETHFees(fees, params.amount)
+    {
+        uint256 length = orders.length;
+        for (uint256 i = 0; i < length; ) {
+            INFTXMarketplaceZap.BuyOrder memory order = orders[i];
+            // Execute fill
+            _buy(
+                orders[i],
+                params.fillTo,
+                params.revertIfIncomplete,
+                order.price
+            );
+            
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+
+    // --- Internal ---
+
+    function _buy(
+        INFTXMarketplaceZap.BuyOrder calldata buyOrder,
+        address receiver,
+        bool revertIfIncomplete,
+        uint256 value
+    ) internal {
+        // Execute the fill
+        try NFTX_MARKETPLACE.buyAndRedeem{value: value}(
+                buyOrder.vaultId, 
+                buyOrder.amount,
+                buyOrder.specificIds,
+                buyOrder.path,
+                receiver
+            )
+        {} catch {
+            // Revert if specified
+            if (revertIfIncomplete) {
+                revert UnsuccessfulFill();
+            }
+        }
+    }
+
+    // --- Single ERC721 offer ---
+
+    function sell(
+        INFTXMarketplaceZap.SellOrder[] calldata orders,
+        OfferParams calldata params,
+        Fee[] calldata fees
+    ) external nonReentrant {
+
+        uint256 length = orders.length;
+        for (uint256 i = 0; i < length; ) {
+            // Execute fill
+            _sell(
+                orders[i],
+                params.fillTo,
+                params.revertIfIncomplete,
+                fees
+            );
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    function _sell(
+        INFTXMarketplaceZap.SellOrder calldata sellOrder,
+        address receiver,
+        bool revertIfIncomplete,
+        Fee[] calldata fees
+    ) internal {
+
+        IERC165 collection = sellOrder.collection;
+        bool isERC721 = collection.supportsInterface(ERC721_INTERFACE);
+        bool isERC1155 = collection.supportsInterface(ERC1155_INTERFACE);
+
+        if (isERC721) {
```

Comment:
Instead of having an `if` / `else` for each part of the code (eg. approval / filling / refunding), let's have a single `if` / `else` with the logic for ERC721 in `if` and the logic for `ERC1155` in the `else`. Also instead of computing `isERC721` and `isERC1155` beforehand let's put them directly in the `if` / `else` conditions so that we don't perform the ERC1155 call only if strictly needed:
```solidity
if (collection.supportsInterface(ERC721_INTERFACE)) {
   // All ERC721 logic in here
} else if (collection.supportsInterface(ERC1155_INTERFACE) {
   // All ERC1155 logic in here
}
```